forte/processors/base/data_selector.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=useless-super-delegation
+
+"""
+A data selector for data augmentation.
+It is a reader that search documents from elastic search indexer
+and yield datapacks.
+An elastic search indexer for data selector needs to be created first.
+Refer to data_augment/data_select_index_pipeline.py for indexer generation.
+"""
+
+from typing import Iterator, Any, Dict, Optional
+
+from forte.common.resources import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.readers.base_reader import PackReader
+from forte.data.data_utils import deserialize
+from forte.indexers.elastic_indexer import ElasticSearchIndexer
+
+
+__all__ = [
+    "BaseElasticSearchDataSelector",
+    "RandomDataSelector",
+    "QueryDataSelector",
+]
+
+
+class BaseElasticSearchDataSelector(PackReader):
```

Comment:
Can we have a base class to be BaseDataSelector, where the actual implementation such as ElasticSearch is not attached here?